### PR TITLE
Add persistent flag to UserExtSource

### PR DESCRIPTION
**Pull request by ondrejvelisek**
from perun.properties file.
Please check UserExtSource model bean. I'm not sure if I should add property to other methods such as toString, serializeToString, equals, hashCode etc. Or should I rewrite tests?

Otherwise it works fine. I've tested it on new WUI user profile.

### CODE REVIEW

File: perun-base/src/main/resources/perun-base.xml [hunk 1]
```diff
@@ -89,6 +89,7 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
 					<prop key="perun.sms.program"></prop>
 					<prop key="perun.readOnlyPerun">false</prop>
 					<prop key="perun.DBInitializatorEnabled">false</prop>
+					<prop key="perun.userExtSources.persistent">PERUN,[\w\d]*</prop>
```
Comment (ondrejvelisek) [inline perun-base/src/main/resources/perun-base.xml]: I guess it is used only in tests.

**Comment by ondrejvelisek** [comment]
Format of property is:  `ExtSourceName1;ExtSourceName2,regex2;ExtSourceName3`
FOr example:  `perun.userExtSources.persistent=PERUN;https://engine.elixir-czech.org/authentication/idp/metadata,.*@elixir-europe.org$`

File: perun-base/src/main/resources/perun-base.xml (hunk 1 above)
Comment (zlamalp) [inline perun-base/src/main/resources/perun-base.xml]: It's used always as a backup, when no other config file is provided to Perun (``/etc/perun/perun.properties``) or such property is missing in it. So it's used by default for in-memory tests.

**Comment by zlamalp** [comment]
For now just skip ``persistent`` param in ``toString()/serializeToString()``. Not everything needs to be in this method. If it will be required, we will add it later.

Also please see ``perun-base`` module, package ``rpc.deserializer`` class ``JsonDeserializer`` and add mixin for ignoring new property (when received by RPC from client apps).

```java
@JsonIgnoreProperties({"persistent"})
private interface UserExtSourceMixIn {}

mapper.getDeserializationConfig().addMixInAnnotations(UserExtSource.class, UserExtSourceMixIn.class);
```

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java [hunk 2]
```diff
@@ -67,6 +72,32 @@
...
+					userExtSourcePersistentPatterns.put(extSource, Pattern.compile(".*"));
+				}
+			}
+		} catch (InternalErrorException e) {
+			log.info("Error when reading property perun.userExtSources.persistent. Serving default behavior.", e);
+			// If error occurred no persistent user ext sources are considered. e.g. property is not set.
+			userExtSourcePersistentPatterns = new HashMap<>();
+		} catch (PatternSyntaxException e) {
+			// cannot use checked exception because of static context
+			throw new InternalErrorRuntimeException("Error when parsing perun.userExtSources.persistent property. Probably regex is in wrong format.", e);
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java]: The PatternSyntaxException can raise, only if "split" argument (which should be regex) is wrongly formatted. In other words, it's not dependent on input (i.e. property from configuration).  
I can guarantee you, that ";" is correct regex :-), therefor this exception never raises.  
Please either delete this "catch" section or at least change the error message in InternalErrorRuntimeException.

Comment (ondrejvelisek) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java]: Done. I've deleted whole catch block.

**Pull request merged**